source/server/converters.h

```diff
@@ -113,6 +114,79 @@ CType convert_from_grpc(const GrpcType& value)
   return static_cast<CType>(value);
 }
 
+inline bool is_ascii(const char* str)
+{
+  for (; *str != '\0'; ++str) {
+    if (static_cast<unsigned char>(*str) > 0x7f) {
+      return false;
+    }
+  }
+  return true;
+}
+
+template <>
+inline std::string convert_from_grpc(const std::string& value)
+{
+  if (is_ascii(value.c_str())) {
+    return value;
+  }
+#if WIN32
+  std::wstring utf16value;
+  utf16value.reserve(value.length());  // often too much, but that's okay
+  utf8::utf8to16(value.begin(), value.end(), std::back_inserter(utf16value));
```

Comment:
Consider: This could be done on Windows using `MultiByteToWideChar(CP_UTF8` since you have to use the Windows methods anyway. It still takes 2 conversions either way. Just a thought.

Reply:
I emulated what others have done in production code. I could research whether it was intentional or not, but since it works correctly like this, it's not worth the effort IMO.